### PR TITLE
Drop old GHC configs and add 8.4.3 to Travis.

### PR DESCRIPTION
**Pull request by robinp**
Also disable the forced unregistering of packages in `build-stack.sh`, since it was not reliably effective, and could also break the package database. Will add doc updates in the future on patterns (either using a clean stack state, or cabal new-build) that achieve proper reindexing.

Fixes Travis after #82.

### CODE REVIEW

**Comment by robinp** [comment]
+ @qrilka, @ivan444

File: README.md [hunk 1]
```diff
@@ -8,10 +8,20 @@ This is not an official Google product.
...
-  - GHC 7.10.3, 8.0.1, 8.0.2, 8.2.1
+  - GHC 8.0.2
+  - GHC 8.2.2
+  - GHC 8.4.3
+
+Stackage:
+  - A recent LTS release corresponding to above compilers is supported.
+    See `stack-ghcXXX.yml` files.
+
+Previous compilers (from GHC 7.10.3) were supported at some point. Look at
```
Comment (ivan444) [inline README.md]: Can we add git tags at the versions which drop a GHC version?
(if people want the older version, they simply checkout that git tag and have everything working)

Comment (robinp) [reply README.md]: Good idea. Created https://github.com/google/haskell-indexer/releases/tag/last-ghc-7.10.3 .

**Comment by qrilka** [comment]
It looks like deleting corresponding .conf file from `.stack-work/install` makes reruns possible but it would be better to have some automated way to do that for packages not from local sources (for which `--force-dirty` does not seem to work)

**Comment by qrilka** [comment]
If it's interesting to anybody the problem I talked above was coming form `setup-config` files tying packages to a particular GHC version (and going around `ghc_kythe_wrapper`)

File: README.md [hunk 2]
```diff
@@ -58,14 +68,14 @@ Use the following to build and run tests:
 ```
 git clone --recursive https://github.com/google/haskell-indexer.git
 cd haskell-indexer
-# 8.17 -> GHC 8.0.2
-# 6.30 -> GHC 7.10.3
-export STACK_YAML=$(readlink -f stack-8.17.yaml)
+export STACK_YAML=$(readlink -f stack-ghc822.yaml)
```
[the remark this answers is not in the document]
Comment (robinp) [reply README.md]: Thanks, updated link! By the way the link is not used in Travis, just there to have something for users who don't explicitly set STACK_YAML.

**Comment by robinp** [comment]
Ah, I lost write access on the repo. @ivan444 @blackgnezdo can you grant me, or would you prefer to merge PRs yourself?

**Comment by ivan444** [comment]
> Ah, I lost write access on the repo. @ivan444 @blackgnezdo can you grant me, or would you prefer to merge PRs yourself?

Seems I can't do that (don't have permissions -- and don't see the repository's settings page).
@blackgnezdo do you have the permission?

**Comment by blackgnezdo** [comment]
I can't either, I was never a contributor to this repo. @ivan444 wanna reach out to get ownership or should I?

**Comment by ivan444** [comment]
I'll do it right away, and add us both.

**Comment by ivan444** [comment]
The permissions are fixed.

**Pull request merged**